soaktest/soaktest.go

```diff
@@ -0,0 +1,258 @@
+package soaktest
+
+/**
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"log"
+	"os"
+	"time"
+)
+
+var DrErrCnt uint64
+var DrCnt uint64
+var producerErrorCbCnt uint64
+var MsgDupCntForOutput uint64
+var MsgMissCntForOutput uint64
+var ProducerMsgId uint64
+var ConsumerMsgCnt uint64
+var ConsumerMsgErrs uint64
+var ConsumerErr uint64
+var ProducerErr uint64
+
+var WarningLogger *log.Logger
+var InfoLogger *log.Logger
+var ErrorLogger *log.Logger
+
+var PRODUCER_DR = "producer.dr"
+var PRODUCER_DR_ERR = "producer.dr.err"
+
+const CONSUMER_ERROR = "consumer.err"
+const PRODUCER_ERROR = "producer.err"
+const FAILED_TO_PRODUCE_MESSAGE = "failed.to.produce.message"
+const CONSUMER_CONSUME_MSG = "consumer.consume.msg"
+const CONSUMER_RECEIVE_ERROR = "consumer.receive.err"
+const PRODUCER_DROK = "producer.drok"
+const LATENCY = "latency"
+const COMSUMER_CONSUME_DUP_MSG = "consumer.consume.dup.msg"
+const COMSUMER_CONSUME_MISS_MSG = "consumer.consume.miss.msg"
+const BROKER_RTT_P99 = "broker.rtt.p99"
+const BROKER_RTT_AVG = "broker.rtt.avg"
+
+func InitLogFiles(path, filename string) {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
```

Comment:
the directory can be derrived from the filename with Dirname(), so the path argument isn't really needed.